state/document.go

```diff
@@ -212,6 +212,7 @@ func loadDocumentAndResetState(state *EditorState, path string, requireExists bo
 	state.documentBuffer.showSpaces = cfg.ShowSpaces
 	state.documentBuffer.autoIndent = cfg.AutoIndent
 	state.documentBuffer.showLineNum = cfg.ShowLineNumbers
+	state.documentBuffer.lineNumberMode = config.LineNumberMode(cfg.LineNumberMode)
```

Comment:
I think we'll also need to update `ReloadDocument` to restore the line number mode in case the user ran the "toggle line number" menu command (similar to what we do for "showLineNum")

Otherwise, this can happen:
1. User opens a document.
2. User toggles the line number mode using the menu command.
3. Some other program updates the document, triggering an auto-reload and resetting the line number mode.